IM_calculation/IM/im_calculation.py

```diff
@@ -507,63 +511,76 @@ def compute_measures_multiprocess(
         components_to_store,
     ) = constants.Components.get_comps_to_calc_and_store(comp)
 
-    bbseries, station_names = get_bbseis(
-        input_path, file_type, station_names, real_only=real_only
-    )
-    total_stations = len(station_names)
-    # determine the size of each iteration base on num of processes and mem
-    steps = get_steps(
-        input_path, process, total_stations, "FAS" in ims and bbseries.nt > 32768
-    )
-
-    # initialize result list for basic IM
-    if not running_adv_im:
-        all_results = []
-
-    with Pool(process) as p, ProgressTracker(total_stations) as p_t:
-        for i in range(0, total_stations, steps):
-            # read waveforms of stations
-            # each iteration = steps size
-            stations_to_run = station_names[i : i + steps]
-            waveforms = read_waveform.read_waveforms(
+    bbseries = None
+    if is_master:
+        bbseries, station_names = get_bbseis(
+            input_path, file_type, station_names, real_only=real_only
+        )
+    bbseries = comm.bcast(bbseries, root=master)
+    station_names = comm.bcast(station_names, root=master)
+
+    # Check which stations to run against non-zero station files already
+    # in the station output directory
+    station_path = Path(output) / "stations"
+    stations_set = set(station_names)
+    found_stations = {
+        file.stem for file in station_path.iterdir() if file.stat().st_size > 0
+    }
+    stations_to_run = list(stations_set.difference(found_stations))
+
+    status = MPI.Status()
+    if is_master:
+        nslaves = size - 1
+        while nslaves:
+            comm.recv(source=MPI.ANY_SOURCE, status=status)
+            slave_id = status.Get_source()
+            # next job
+            if len(stations_to_run) > 0:
+                station = stations_to_run.pop(-1)
+                logger.info(f"Sending station {station} to {slave_id}")
+                comm.send(obj=station, dest=slave_id)
+            else:
+                comm.send(obj=StopIteration, dest=slave_id)
+                nslaves -= 1
+        logger.info("All stations complete")
+    else:
+        for station in iter(lambda: comm.sendrecv(None, dest=master), StopIteration):
```

Comment:
The last command sent is stop iteration, does it need to be in the worker for loop too?